### PR TITLE
add impala algorithm in paddle

### PR DESCRIPTION
**Pull request by zenghsh3**

### CODE REVIEW

File: examples/IMPALA/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+### Distributed Training:
+
+At first, We can start a local cluster with 32 CPUs:
+
+```bash
+xparl start --port 8010 --cpu_num 32
+```
+
+Note that if you have started a master before, you don't have to run the above
```
Comment (TomorrowIsAnOtherDay) [inline examples/IMPALA/README.md]: Note that it is not necessary to run the command each time before training. We can reuse the xparl cluster for distributed training if we have started it before.

File: examples/IMPALA/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+## Reproduce IMPALA with PARL
+Based on PARL, the IMPALA algorithm of deep reinforcement learning is reproduced, and the same level of indicators of the paper is reproduced in the classic Atari game.
+
+> Paper: IMPALA in [Impala: Scalable distributed deep-rl with importance weighted actor-learner architectures](https://arxiv.org/abs/1802.01561)
+
+### Atari games introduction
+Please see [here](https://gym.openai.com/envs/#atari) to know more about Atari games.
+
+### Benchmark result
+Result with one learner (in a P40 GPU) and 32 actors (in 32 CPUs).
```
Comment (TomorrowIsAnOtherDay) [inline examples/IMPALA/README.md]: Result-> Learning curve.

File: examples/IMPALA/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
++ PongNoFrameskip-v4: mean_episode_rewards can reach 18-19 score in about 10 minutes.
+<img src="https://github.com/benchmarking-rl/PARL-experiments/raw/master/IMPALA/Pong.png" width = "400" height ="300" alt="IMPALA_Pong" />
+
++ Results(mean_episode_rewards) of other games in an hour.
```
Comment (TomorrowIsAnOtherDay) [inline examples/IMPALA/README.md]: Results-> Learning curves

File: examples/IMPALA/atari_agent.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import parl
+from parl.utils.scheduler import PiecewiseScheduler, LinearDecayScheduler
+import paddle
+
+
+class AtariAgent(parl.Agent):
+    def __init__(self, algorithm, obs_shape, act_dim):
+        assert isinstance(obs_shape, (list, tuple))
+        assert isinstance(act_dim, int)
+        self.obs_shape = obs_shape
```
Comment (TomorrowIsAnOtherDay) [inline examples/IMPALA/atari_agent.py]: This attribute has not been used in the class.

File: examples/IMPALA/atari_agent.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        self.act_dim = act_dim
```
Comment (TomorrowIsAnOtherDay) [inline examples/IMPALA/atari_agent.py]: This attribute has not been used in the class.

File: examples/IMPALA/atari_agent.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (TomorrowIsAnOtherDay) [inline examples/IMPALA/atari_agent.py]: 2018->2022. Same problem in other file headers.

**Comment by ShuaibinLi** [comment]
A new PR(https://github.com/PaddlePaddle/PARL/pull/993) for IMPALA-paddle is merged.

**Pull request closed without merging**